### PR TITLE
[microtvm][Zephyr] Add project overlay to overwrite device tree configs

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR adds a mechanism to overwrite device tree configs in zephyr project. It also adds an overlay for nucleo_l4r5zi board to change processor frequency from 80MHz which is the default to 120MHz.

cc @alanmacd @areusch @gromero @guberti

### CODE REVIEW

**Review by areusch — APPROVED** [review]
Thanks @mehrdadh

**Review by guberti — APPROVED** [review]
LGTM - thanks for this change!

**Comment by gromero** [comment]
@mehrdadh Nice findings, I would if you would like to upstream it to Zephyr also so other projects could pick this up?

File: apps/microtvm/zephyr/template_project/app-overlay/nucleo_l4r5zi.overlay [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+&rcc {
+	clock-frequency = <DT_FREQ_M(120)>;
+};
```
Comment (erwango) [inline apps/microtvm/zephyr/template_project/app-overlay/nucleo_l4r5zi.overlay:23]: That is not correct.
You should update the PLL configuration accordingly:
```
&pll {
	div-m = <2>;
	mul-n = <30>;
	div-p = <7>;
	div-q = <2>;
	div-r = <2>;
	clocks = <&clk_hsi>;
	status = "okay";
};
```
Note that the impacts of effects on bus configurations have to be tested.

Comment (gromero) [reply apps/microtvm/zephyr/template_project/app-overlay/nucleo_l4r5zi.overlay:23]: @erwango Thanks a lot. Do you mind to tell how exactly one finds the params (M, N, P, Q, and R) for the PLL? I see them in the clock tree in  RM0432 rev. 9 (Arm Reference Manual), Figure 16, but could not find any formula about how to relate them to the 120 MHz freq.

Comment (erwango) [reply apps/microtvm/zephyr/template_project/app-overlay/nucleo_l4r5zi.overlay:23]: > @erwango Thanks a lot. Do you mind to tell how exactly one finds the params (M, N, P, Q, and R) for the PLL? I see them in the clock tree in RM0432 rev. 9 (Arm Reference Manual), Figure 16, but could not find any formula about how to relate them to the 120 MHz freq.

Core freq = Input freq (16MHz) / M(2) * N(30) / R(2)

Comment (gromero) [reply apps/microtvm/zephyr/template_project/app-overlay/nucleo_l4r5zi.overlay:23]: @erwango Thanks for the explanation. I found also the formula in the Reference Manual, below the bit descriptions for the `RCC_PLLCFGR`, where also there are formulas using the Q and P params. Could you please review https://github.com/apache/tvm/pull/12756 ?

**Comment by mehrdadh** [comment]
@erwango thanks for catching this and @gromero for the follow up PR!

**Pull request merged**